MetaModelica external function calls desctructors missing from generated C code

### Description

When generating C code for MetaModelica newer C compilers complain about missing function declarations for some external destructor functions.

```
ISO C99 and later do not support implicit function declarations
```

### Steps to Reproduce

Run `testsuite/openmodelica/bootstrapping/ExpressionTest.mos`.
This needs OpenModelica build with the Autoconf + Makefile build, since some MetaModelica code expects some CodegenC t o be in a specific location.


```bash
$ omc ExpressionTest.mos

"Error: Error building simulator. Build log: /usr/bin/clang -O0 -DOM_HAVE_PTHREADS -fPIC -DOM_HAVE_PTHREADS   \"-I/home/andreas/workdir/OpenModelica/OMCompiler/Compiler/Util/\" -I\"/home/andreas/workdir/OM/OpenModelicaMaster/build_cmake/install_cmake/bin/../include/omc/c\" -I\"/home/andreas/workdir/OM/OpenModelicaMaster/build_cmake/install_cmake/bin/../include/omc\"  -c -o ExpressionTest_test.o ExpressionTest_test.c
[...]
ExpressionTest_test.c:15974:3: error: call to undeclared function 'omc_File_File_destructor'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 15974 |   omc_File_File_destructor(threadData,_file);
       |   ^
ExpressionTest_test.c:15974:3: note: did you mean 'omc_File_File_constructor'?
./ExpressionTest_test.h:1974:18: note: 'omc_File_File_constructor' declared here
 1974 | modelica_complex omc_File_File_constructor(threadData_t *threadData, modelica_metatype _fromID);
      |                  ^
ExpressionTest_test.c:16203:3: error: call to undeclared function 'omc_File_File_destructor'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 16203 |   omc_File_File_destructor(threadData,_file);
       |   ^
ExpressionTest_test.c:17956:3: error: call to undeclared function 'omc_File_File_destructor'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 17956 |   omc_File_File_destructor(threadData,_file);
       |   ^
ExpressionTest_test.c:19022:3: error: call to undeclared function 'omc_File_File_destructor'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 19022 |   omc_File_File_destructor(threadData,_file);
       |   ^
ExpressionTest_test.c:41086:3: error: call to undeclared function 'omc_System_StringAllocator_destructor'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 41086 |   omc_System_StringAllocator_destructor(threadData,_sb);
       |   ^
ExpressionTest_test.c:41086:3: note: did you mean 'omc_System_StringAllocator_constructor'?
./ExpressionTest_test.h:3468:18: note: 'omc_System_StringAllocator_constructor' declared here
 3468 | modelica_complex omc_System_StringAllocator_constructor(threadData_t *threadData, modelica_integer _sz);
      |                  ^
[...]
10 warnings and 5 errors generated.
make: *** [ExpressionTest_test.makefile:25: ExpressionTest_test] Error 1
"
```

### Expected Behavior

In addition to 

```C
DLLDirection
modelica_complex omc_File_File_constructor(threadData_t *threadData, modelica_metatype _fromID);
DLLDirection
modelica_metatype boxptr_File_File_constructor(threadData_t *threadData, modelica_metatype _fromID);
static const MMC_DEFSTRUCTLIT(boxvar_lit_File_File_constructor,2,0) {(void*) boxptr_File_File_constructor,0}};
#define boxvar_File_File_constructor MMC_REFSTRUCTLIT(boxvar_lit_File_File_constructor)
```

the generated code needs function declarations for `omc_File_File_destructor` and `omc_System_StringAllocator_destructor`.

### Version and OS

<!--- Please complete the following information. -->

 - OpenModelica Version: OMCompiler v1.26.0-dev.481+g7da1f253a1
 - OS: Ubuntu Focal
 - clang: Ubuntu clang version 18.1.3 (1ubuntu1)

### Additional Context

This is an error for all newer versions of C copilers (clang, maybe gcc as well)

--- a/OMCompiler/Compiler/SimCode/SimCodeFunctionUtil.mo
+++ b/OMCompiler/Compiler/SimCode/SimCodeFunctionUtil.mo
@@ -2458,7 +2458,7 @@ protected function addDestructor2
   output HashTableStringToPath.HashTable ht = inHt;
 algorithm
   if not BaseHashTable.hasKey(pathstr, ht) then
-    BaseHashTable.add((pathstr, path), ht);
+    ht := BaseHashTable.add((pathstr, path), ht);
   end if;
 end addDestructor2;
 

